src/renderer/features/proxied-wallet/service.ts

```diff
@@ -1,238 +1,137 @@
-//     return {
-//       type: 'proxy',
-//       name: 'Any',
-//     };
-//   }
-//
-//   if (proxyType === 'NonTransfer' && call !== 'Balances') {
-//     return {
-//       type: 'proxy',
-//       name: 'NonTransfer',
-//     };
-//   }
-//
-//   if (proxyType === 'Staking' && Staking.includes(call)) {
-//     return {
-//       type: 'proxy',
-//       name: 'Staking',
-//     };
-//   }
-//
-//   if (proxyType === 'NominationPools' && NominationPools.includes(call)) {
-//     return {
-//       type: 'proxy',
-//       name: 'NominationPools',
-//     };
-//   }
-//
-//   if (proxyType === 'Auction' && Auction.includes(call)) {
-//     return {
-//       type: 'proxy',
-//       name: 'Auction',
-//     };
-//   }
-//
-//   if (proxyType === 'Governance' && Governance.includes(call)) {
-//     return {
-//       type: 'proxy',
-//       name: 'Governance',
-//     };
-//   }
-//
-//   if (proxyType === 'CancelProxy' && CancelProxy.includes(call)) {
-//     return {
-//       type: 'proxy',
-//       name: 'CancelProxy',
-//     };
-//   }
-//
-//   if (proxyType === 'IdentityJudgement' && IdentityJudgement.includes(call)) {
-//     return {
-//       type: 'proxy',
-//       name: 'IdentityJudgement',
-//     };
-//   }
-//
-//   return null;
-// }
-
-// function narrowProxyCall(proxyCall: ProxyCall, proxyType: ProxyType): ProxyCall | null {
-//   if (isSuperset(proxyCall.name, proxyType)) {
-//     return {
-//       type: 'proxy',
-//       name: proxyType,
-//     };
-//   }
-//   return null;
-// }
-
-// function isSuperset(x: ProxyType, y: ProxyType) {
-//   if (x === y) return true;
-//   if (x === 'Any') return true;
-//   if (y === 'Any') return false;
-//   if (x === 'NonTransfer') return true;
-//   return false;
-// }
+/**
+ * Check if call can be executed with given proxy type
+ *
+ * TODO can be replaced with papi view function call after adoption
+ *
+ * ```ts
+ * const res = await api.view.Proxy.check_permissions(
+ *   extrinsic.decodedCall,
+ *   { type: proxyType, value: undefined },
+ * );
+ * ```
+ */
+function checkCallPermission(proxyType: ProxyType, call: string): boolean {
+  // known pallets that can be checked by hand.
+  // this list may extend later, but it will be perfect if polkadot sdk will implement permission checks with query
+  const Staking: string[] = ['utility', 'staking', 'session', 'fastUnstake', 'voterList', 'nominationPools'];
+  const NominationPools: string[] = ['utility', 'nominationPools'];
+  const CancelProxy: string[] = ['proxy'];
+  const Auction: string[] = ['auctions', 'crowdloan', 'registrar', 'slots'];
+  const IdentityJudgement: string[] = ['identityJudgement'];
+  const Governance: string[] = [
+    'utility',
+    'treasury',
+    'bounties',
+    'childBounties',
+    'convictionVoting',
+    'referenda',
+    'whitelist',
+  ];
+
+  if (proxyType === 'Any') {
+    return true;
+  }
+
+  if (proxyType === 'NonTransfer') {
+    return call !== 'balances';
```

Comment:
NonTransfer it is not only balances:

https://github.com/paritytech/polkadot-sdk/blob/fd417de617f303b84a0cac1972cf5d7090000d2f/cumulus/parachains/runtimes/assets/asset-hub-westend/src/lib.rs#L698